### DISCUSSION TITLE
Inline middleware not working as expected if the inline element has lines more than 2?

### DISCUSSION THREAD

**Discussion by Zzzen** [floating-ui/floating-ui#1790] [General]
![image](https://user-images.githubusercontent.com/6630042/179147058-0110d2f0-ae15-44cf-9303-2a05a4cf5050.png)

As shown above, the blue text is an inline anchor element. When I move the cursor to `that`, I would expect tooltip is shown above the middle of the second line. But now it is shown above the first line.

https://github.com/floating-ui/floating-ui/blob/master/packages/core/src/middleware/inline.ts#L77-L101

**Comment by atomiks** [comment]
Why would that be expected? If you hovered over the last line would you expect it to be placed there or not because it overlaps the middle text? If the first line extends enough toward "that", it will also get overlapped in this scenario

**Comment by cf-kialo** [comment]
I agree, the documentation says "By default, inline() infers which of the ClientRects to choose based on the placement. However, you may want a different rect to be chosen.".
But no matter how much you change x and y, as soon as there are more than 2 lines it always picks the first rect as reference.
My usecase is a link the user could possibly paste that spans over many lines. On click, it should open a tooltip to edit the link. If that always appears at the very top of the link it might be out of view. So for calculating the tooltip position I would like floatingui to treat the clicked client rect as reference element.
